### PR TITLE
ci: add github-actions ci lint, coverage and nightly

### PR DESCRIPTION
**Pull request by panva**
Adds [Github Actions](https://github.com/features/actions) CI with the following jobs

- lint
- test matrix on all actions-available platforms + minimal and current node versions
- test on nightly

[example run](https://github.com/panva/webcrypto/commit/0a09e9f0f721a549111ff12aec5599ae3d39a53e/checks)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: npm install
+      - run: npm run lint
+
+  test:
+    name: Node Tests + Coverage
```
Comment (tniessen) [inline .github/workflows/ci.yml]: Tiny nit: `Node.js`. We use `Node.js` and `node`, but never `node.js` or `Node`.

**Comment by tniessen** [comment]
Do we have any rules or guidelines regarding GitHub actions @nodejs/tsc?

**Comment by Trott** [comment]
> Do we have any rules or guidelines regarding GitHub actions @nodejs/tsc?

I think @nodejs/build may be experimenting with them in places? AFAIK, there are no guidelines or rules. This seems OK to me, but perhaps superfluous because you already have Travis? Although maybe it does stuff Travis doesn't.

Anyway, I defer to others on this, but I don't have a problem with it.

**Comment by cclauss** [comment]
My sense is that GitHub Actions is more modern than Travis CI in many ways and more pleasant to work with but it remains in Beta until November so there is still quirkiness.ness

**Comment by rvagg** [comment]
no guidelines and not much experimentation (if any?) in the org. I'd encourage experimentation like this because it's bound to become standard over time in the org and the sooner we figure out best practices the better. +1

**Comment by cclauss** [comment]
Other Actions experiments:
* nodejs/node#29474 (open)
* nodejs/node#29193 (closed)
* nodejs/node-gyp#1671 (closed)

**Comment by rvagg** [comment]
Very sorry @cclauss, thanks for the correction

**Comment by tniessen** [comment]
Sorry @panva, I forgot about this. Would you mind updating the minimum Node.js version to match `package.json` so I can land this?

**Comment by panva** [comment]
@tniessen done, it'll take a merge to master to see the action run tho.

**Pull request merged**